internal/rbd/driver/driver.go

```diff
@@ -219,7 +219,7 @@ func (r *Driver) setupCSIAddonsServer(conf *util.Config) error {
 		fcs := casrbd.NewFenceControllerServer()
 		r.cas.RegisterService(fcs)
 
-		rcs := casrbd.NewReplicationServer(NewControllerServer(r.cd))
+		rcs := casrbd.NewReplicationServer(rbd.CSIInstanceID, NewControllerServer(r.cd))
```

Comment:
@Madhu-1 , shouldn't this be
```suggestion
rcs := casrbd.NewReplicationServer(conf.InstanceID, NewControllerServer(r.cd))
```
?

Reply:
yes, I think so. Can you send a PR for that?

Reply:
`conf.InstanceID` will not work if its not set, we need to use default in that case we are not updating conf.InstanceID with rbd.CSIInstanceID, that's the reason to use `rbd.CSIInstanceID` as its gets updated always or it contains the default

Reply:
@Madhu-1, we are doing that for InitJournals - 
https://github.com/ceph/ceph-csi/blob/db6f54f75d87261d66cada71e6a2a0f8665cec18/internal/rbd/globals.go#L93-L100

And `conf.InstanceID` is passed to InitJournals

Reply:
if the conf.InsatceID is empty its not updated with the default one that we have in the code right rbd.InitJournals(conf.InstanceID)? with this one https://github.com/ceph/ceph-csi/blob/db6f54f75d87261d66cada71e6a2a0f8665cec18/internal/rbd/globals.go#L28

Reply:
If `conf.Instance` is empty,  `CSIInstanceID` (default) is used for journals. Check the above code I have linked in previous comment.

Reply:
Apologies, I was mistaken it was other way around. 
Thanks @Madhu-1.

Then I guess for VolumeGroupServer we need to use `rbd.CSIInstanceID` 
https://github.com/ceph/ceph-csi/blob/db6f54f75d87261d66cada71e6a2a0f8665cec18/internal/rbd/driver/driver.go#L223
@nixpanic ?

Reply:
`rbd.CSIInstanceID` seems to be a global value, it is configurable on the commandline, so I think using the configured value is cleaner (all config options should have default values anyway?).

In `internal/rbd/driver/driver.go` there is `rbd.InitJournals(conf.InstanceID)`, so in the end it is all the same.

Reply:
`conf.InstanceID` default value is empty and `rbd.CSIInstanceID` is replaced with `conf.InstanceID` when it is configured in command line.
So using `conf.InstanceID` when its not configured  will lead to creating journal directory `csi.groups.` ?

Reply:
yeah, I noticed that too. The global `rbd.CSIInstanceID` could be set to something earlier on, in case the ordering of (seemingly unrelated) initialization of the journals is done later.  This is a little fragile, and using global variables is quite ugly in any case. I'm preparing a PR for improving it.

Reply:
see #4747